File: MacTemplet/Cagtegory/NSMenu+Helper.m


//
//  NSMenu+Helper.m
//  MacTemplet
//
//  Created by Bin Shang on 2019/6/20.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

#import "NSMenu+Helper.h"
#import "NSMenuItem+Helper.h"

@implementation NSMenu (Helper)

- (NSMenuItem *)insertItemWithTitle:(NSString *)string keyEquivalent:(NSString *)charCode atIndex:(NSInteger)index handler:(void(^)(NSMenuItem *menuItem))handler{
    NSMenuItem * item = [self addItemWithTitle:@"Load1" action:nil keyEquivalent:charCode];
    [item addActionHandler:handler];
    return item;
}

- (NSMenuItem *)addItemWithTitle:(NSString *)string keyEquivalent:(NSString *)charCode handler:(void(^)(NSMenuItem *menuItem))handler{
    NSMenuItem * item = [self addItemWithTitle:string action:nil keyEquivalent:charCode];
    [item addActionHandler:handler];
    return item;
}

@end

File: MacTemplet/MacCagtegory/NSMenu+Helper.swift
//
//  NSMenu+Helper.swift
//  MacTemplet
//
//  Created by Bin Shang on 2019/11/22.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

import Cocoa

@objc public extension NSMenu {

    func addItem(withTitle title: String, keyEquivalent: String, handler: @escaping (NSMenuItem) -> Void) -> NSMenuItem {
        let item = addItem(withTitle: title, action: nil, keyEquivalent: keyEquivalent)
        item.addAction(handler)
        return item;
    }

    func insertItem(withTitle title: String, keyEquivalent charCode: String, at index: Int, handler: @escaping (NSMenuItem) -> Void) -> NSMenuItem {
        let item = insertItem(withTitle: title, action: nil, keyEquivalent: charCode, at: index)
        item.addAction(handler)
        return item;
    }
}
